pep-0678.rst

```diff
@@ -140,10 +140,10 @@ are collecting multiple exception objects to handle together. [1]_
 Specification
 =============
 
-``BaseException`` gains a new read-only attribute ``__notes__``, an initially
-empty tuple, and a new method ``.add_note(note: str)``.  ``note`` is added to
-the exception's notes, which appear in the standard traceback after the
-exception string. A ``TypeError`` is raised if ``note`` is not a string.
+``BaseException`` gains a new new method ``.add_note(note: str)``. Notes are
+exposed as a tuple via the read-only attribute ``__notes__``, and appear in
+the standard traceback after the exception string.  ``.add_note()`` raises
+``TypeError`` if ``note`` is not a string.
```

Comment:
Hmm, that's a little surprising because it makes the obvious test for `ExceptionGroup.derive()` (`a.__notes__ is b.__notes__`) fail.  Would it be much more annoying to have `add_note()` do the moral equivalent of `e.__dict__["__notes__"] = e.__notes__ + (note,)`?

(I assume you mean *access*, not *call*)

Reply:
I can do that. Then to answer Brett’s question the pep needs to say that a new tuple is created when we add/clear notes.

Reply:
OK, let's go with that approach.

(I know it's *possible* to mutate tuples, but it's still pretty weird, and would prevent a pure-Python backport)

Reply:
Sure, I’m not suggesting we do that, just that Brett’s point was that the pep should be clear that we’re not doing it.

Reply:
I've updated the PR now.

Reply:
This is a tangent, but I think even in C you wouldn't be able to grow a tuple in place; https://docs.python.org/3.10/c-api/tuple.html#c._PyTuple_Resize exists but to grow the tuple it has to reallocate it in a different place.